# Titanic Survival Analysis

## Business Objective

The objective of this project is to analyze the Titanic passenger dataset to discover patterns related to passenger demographics, ticket classes, fares, and survival outcomes using Python and Pandas.

In [40]:
#import our pandas 
import pandas as pd

### Data Inspection 

In [28]:
#loading our raw csv datset titanic.csv
df = pd.read_csv("titanic.csv")

In [30]:
#inspection for the first 5 row of our dataset
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [29]:
#All info about the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [ ]:
#How many row and column is in the dataset?
df.shape

In [13]:
#To get familiar with the column on the dataset
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [ ]:
#Statistical Summary of the dataset
df.describe()

### Data Quality check 

In [ ]:
#quality check any missing values & the total?
df.isnull().sum()

In [ ]:
#any duplicates values & the total ?
df.duplicated().sum()

### Analysis & Insight

#### Question 1
How many passengers survived?

In [ ]:
df["Survived"].value_counts()

549 passengers did not survive, while 342 passengers survived. This indicates that the majority of passengers on the Titanic did not survive the disaster.

#### Question 2

Did gender affect survival?

In [ ]:
df.groupby("Sex")["Survived"].mean()

The analysis shows that female passengers had a much higher survival rate than male passengers. Approximately 74% of female passengers survived, compared to about 19% of male passengers. This suggests that gender was strongly associated with survival during the Titanic disaster.

#### Question 3
Did passenger class affect survival?

In [23]:
df.groupby("Pclass")["Survived"].mean()

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64

The analysis shows that First Class passengers had the highest survival rate (approximately 63%), followed by Second Class passengers (about 47%). Third Class passengers had the lowest survival rate (approximately 24%). This suggests that passenger class had a significant influence on survival during the Titanic disaster.

#### Question 4 
Did age affect survival?

In [25]:
df.groupby("Survived")["Age"].mean()

Survived
0    30.626179
1    28.343690
Name: Age, dtype: float64

The analysis indicates that passengers who survived were, on average, slightly younger than those who did not survive. Survivors had an average age of approximately 28 years, while non-survivors had an average age of about 31 years. This suggests that age may have had some influence on survival, although the difference is relatively small compared to factors such as passenger class and gender.

#### Question 5

Did the fare paid affect survival?

In [27]:
df.groupby("Survived")["Fare"].mean()

Survived
0    22.117887
1    48.395408
Name: Fare, dtype: float64

The analysis indicates that passengers who survived paid higher fares on average than those who did not survive. Survivors paid an average fare of approximately 48, while non-survivors paid an average fare of about 22. This suggests that passengers who purchased more expensive tickets were more likely to survive the Titanic disaster.

#### Question 6

Did passengers traveling with siblings or spouses have a higher survival rate?

In [34]:
df.groupby("SibSp")["Survived"].mean()

SibSp
0    0.345395
1    0.535885
2    0.464286
3    0.250000
4    0.166667
5    0.000000
8    0.000000
Name: Survived, dtype: float64

The analysis shows that survival rates varied across different SibSp groups. Passengers traveling with one sibling or spouse appeared to have a higher survival rate than some other groups, while those traveling in larger family groups had lower survival rates. This suggests that the number of siblings or spouses accompanying a passenger may have influenced survival, although the relationship is not consistent across all groups.

#### Question 7

Did passengers traveling with parents or children have a higher survival rate?

In [33]:
df.groupby("Parch")["Survived"].mean()

Parch
0    0.343658
1    0.550847
2    0.500000
3    0.600000
4    0.000000
5    0.200000
6    0.000000
Name: Survived, dtype: float64

The analysis indicates that passengers traveling with one, two, or three parents or children generally had higher survival rates than those traveling without parents or children. Passengers with three parents or children had the highest survival rate (60%), followed by those with one (55.1%) and two (50%). In contrast, passengers traveling without parents or children had a survival rate of approximately 34.4%, while those traveling with four or six parents or children did not survive. These findings suggest that traveling with a small family group may have been associated with a higher chance of survival, whereas very large family groups experienced lower survival rates.

##### Question 8

Which embarkation port had the highest survival rate?

In [32]:
df.groupby("Embarked")["Survived"].mean()

Embarked
C    0.553571
Q    0.389610
S    0.336957
Name: Survived, dtype: float64

The analysis indicates that the embarkation port was associated with different survival rates among passengers. Passengers who boarded the Titanic at Cherbourg (C) had the highest survival rate (approximately 55.4%), followed by Queenstown (Q) at about 39.0%. Southampton (S) had the lowest survival rate (approximately 33.7%). These findings suggest that the port of embarkation may have influenced survival outcomes, although the dataset does not provide enough information to explain the reasons for these differences.

#### Question 9

Did passengers with an assigned cabin have a higher survival rate?

In [46]:
df.groupby(df["Cabin"].notnull())["Survived"].mean()

Cabin
False    0.299854
True     0.666667
Name: Survived, dtype: float64

The analysis shows that passengers with assigned cabin have a higher survival rate with (approximately 66%) rate while the passengers without assigned cabin had the lower survival rate with (approximately 29%). These findings show that cabin assignment influence survival outcomes.

#### Question 10

Which passenger class paid the highest average fare?

In [48]:
df.groupby("Pclass")["Fare"].mean()

Pclass
1    84.154687
2    20.662183
3    13.675550
Name: Fare, dtype: float64

The analysis shows that First Class passengers paid the highest average fare (approximately 84.15), followed by Second Class passengers (approximately 20.66). Third Class passengers paid the lowest average fare (approximately 13.68). These findings indicate that ticket prices increased with passenger class, with First Class being the most expensive and Third Class the least expensive.

# Titanic Survival Analysis

## Executive Summary

The objective of this project is to analyze the Titanic dataset to identify the factors that influenced passenger survival during the disaster. Using Python and the Pandas library, the dataset was inspected, cleaned, and analyzed to answer key business questions related to passenger demographics, ticket information, family relationships, and travel characteristics. The findings provide insights into the patterns associated with survival and demonstrate fundamental data analysis techniques.

## Key Findings

- Out of 891 passengers, 342 survived while 549 did not survive.
- Female passengers had a significantly higher survival rate than male passengers.
- First Class passengers had the highest survival rate, while Third Class passengers had the lowest.
- Survivors were slightly younger on average than non-survivors.
- Passengers who paid higher fares generally had higher survival rates.
- Passengers traveling with one sibling or spouse had the highest survival rate, while very large family groups had lower survival rates.
- Passengers traveling with one to three parents or children generally had higher survival rates than those traveling alone.
- Passengers who boarded the ship at Cherbourg had the highest survival rate among the three embarkation ports.
- Passengers with assigned cabin numbers had substantially higher survival rates than those without assigned cabins.
- First Class passengers paid the highest average ticket fare.

## Business Recommendations

Based on the analysis, the following recommendations can be made:

- Improve emergency evacuation procedures to ensure equal access for passengers across all travel classes.
- Provide clearer emergency guidance and support for passengers traveling in larger family groups.
- Ensure cabin assignment and passenger location do not disadvantage any group during emergency situations.
- Continue reviewing boarding and evacuation procedures to reduce differences in survival outcomes among passenger groups.
- Collect more detailed passenger information in future datasets to better understand the factors influencing survival.

## Conclusion

This analysis identified several factors associated with passenger survival during the Titanic disaster. Gender, passenger class, ticket fare, family size, embarkation port, and cabin assignment all showed meaningful relationships with survival outcomes. While the analysis identifies patterns within the dataset, it does not establish cause-and-effect relationships. Overall, this project demonstrates how exploratory data analysis can be used to uncover meaningful insights and support evidence-based decision-making.